scipy/stats/tests/test_stats.py

```diff
@@ -1179,6 +1179,14 @@ def test_weightedtau():
     tau, p_value = stats.weightedtau(x, y)
     assert_approx_equal(tau, -0.56694968153682723)
 
+def test_segfault_issue_9710():
+    # https://github.com/scipy/scipy/issues/9710
+    # This test was created to check segfault
+    # In issue SEGFAULT only repros in optimized builds after calling the function twice
+    stats.weightedtau([1], [1.0])
+    stats.weightedtau([1], [1.0])
+    # The code below also caused SEGFAULT
+    stats.weightedtau([np.nan], [52])
```

Comment:
This works but is a bit weird for a test as it doesn't provide an assertion. I realize a segfault isn't a standard thing to test. I had to think about this one for a bit but I think this would work: 

```
segfaultingCommand = subprocess.run(["python",  "-c", "import scipy.stats as ss; from numpy import nan; ss.weightedtau([nan], [52])"])
assert segfaultingCommand.returncode == 0
```

I think that should work (after your fix is added of course) I didn't checkout this branch so I haven't tested w/your fix but it gives a `-11` return code on my mac, the actual return code might vary depending on OS, but I think `0` is a safe exit status on most linux, I think that would be safe for windows too but I don't have a windows box.

Reply:
of course you'd need to add an import for `subprocess` in the  `test_stats.py` file too.

Reply:
I think we use this quite often to test for segfaults. (Without the complexity of starting a subprocess etc)

Reply:
1. The subprocess call also has to make assumptions about the path (and more), adding to it's complexity. Not having to resort to it is attractive.

2. Does `_toint64` work correctly If `x` has no elements? Or is it never called with an empty array?

Reply:
_toint64 is never called with an empty array, because before calling it, `stats.weightedtau` checks whether the arrays are empty and returns `WeightedTauResult(correlation=nan, pvalue=nan)`  if so

Reply:
Agreed that this is the correct way to test for a segfault.